### PR TITLE
feat: add ci and cd actual code

### PR DESCRIPTION
**Pull request by cristianmtr**
Depends on findings from https://github.com/jina-ai/executor-simpleindexer/pull/3

### CODE REVIEW

File: workflow-templates/executor-ci.yml [hunk 1]
```diff
@@ -1,17 +1,193 @@
...
+      - name: Run unit tests
+        run: pytest -s -v -m "not gpu"
+      # - name: Print image size
+      #   run: |
+      #     IMG_NAME=$(echo "${PWD##*/}" | tr '[:upper:]' '[:lower:]')
+      #     docker inspect $IMG_NAME | jq '.[0].Size' | numfmt --to=iec --suffix=B
+
+  # @@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@
+  # @@@@@@@ ⚠️ DEPRECATED ⚠️ @@@@@@@@@
+  # @@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@
```
Comment (maximilianwerk) [inline workflow-templates/executor-ci.yml:91]: I think when we add this CI/CD for single executor repos, we should remove this and make sure any newly created repo does not depend on it.

Comment (cristianmtr) [reply workflow-templates/executor-ci.yml:91]: Will adapt after trying it out on one executor repo jina-ai/executor-simpleindexer#3

**Comment by cristianmtr** [comment]
@maximilianwerk recheck

**Review by maximilianwerk — APPROVED** [review]
LGTM. Nice step towards a better world of Executors :)

**Pull request merged**